guide/popular-topics/components-v2.md

```diff
@@ -0,0 +1,228 @@
+
+To use the new **Components V2 (CV2)** components, you need to pass in the new `IsComponentsV2` message flag from the <DocsLink path="MessageFlags:Enum" /> enum when sending a message. The flag should only be added to the message's `flags` field when the message contains CV2 components, unlike the `Ephemeral` message flag that can be added when you defer an interaction response.
+
+::: danger
+Once a message is sent or edited with the `IsComponentsV2` message flag, the flag **cannot** be removed from that message.
+:::
+
+All components, both new and existing, now have a new field `id` (which should not be confused with the existing `custom_id` field for interactive components), which is an optional 32-bit integer identifier for a component presented in a message. It is used to identify non-interactive components in the response from an interaction. More information about these can be found [here](https://discord.com/developers/docs/components/reference#anatomy-of-a-component).
+
+In the following section, we will explain what kind of components are new, how they work together with interactive components, and the limitations Discord has set when using CV2 components in your message.
+
+## New components
+
+CV2 brought new layout components and some new content components. The content components are very similar to some elements used in embeds, but their behaviour is slightly different when used relative to their use in embeds.
+
+### Text Display
+
+A Text Display is a content component allowing you to add text to your message formatted with Discord's available markdown and the mention of users and roles (which actually notifies the users and roles, instead of the text in embeds!). This component is very similar to the `content` field of a message, but using multiple of these components you can control the layout of your message. You can use the <DocsLink path="TextDisplayBuilder:Class" /> utility class to easily create a Text Display component.
+
+The example below shows how you can send a Text Display component in a channel.
+
+```js
+const { TextDisplayBuilder, MessageFlags } = require('discord.js');
+
+const exampleTextDisplay = new TextDisplayBuilder()
+	.setContent('This text is inside a new Text Display component! You can use **any __markdown__** available inside this component too.');
+
+await channel.send({
+	components: [exampleTextDisplay],
+	flags: MessageFlags.IsComponentsV2,
+});
+```
+
+### Thumbnail
+
+A Thumbnail is a content component that is visually similar to the `thumbnail` field inside an embed, but with CV2 it can only be added as an accessory inside a [Section](/popular-topics/components-v2.md#section) component. However, you can add ALT text to the image as well as marking the image as spoiler. You can use the <DocsLink path="ThumbnailBuilder:Class" /> utility class to easily create a Thumbnail component.
```

Comment:
```suggestion
A Thumbnail is a content component that is visually similar to the `thumbnail` field inside an embed, but with CV2, it can only be added as an accessory inside a [Section](/popular-topics/components-v2.md#section) component. However, you can add alt text to the image as well as marking the image as a spoiler. You can use the <DocsLink path="ThumbnailBuilder:Class" /> utility class to easily create a Thumbnail component.
```